src/ripple/resource/impl/ResourceManager.cpp

```diff
@@ -18,90 +18,110 @@
 //==============================================================================
 
 #include <BeastConfig.h>
-#include <ripple/basics/chrono.h>
 #include <ripple/resource/ResourceManager.h>
+#include <ripple/basics/chrono.h>
+#include <ripple/basics/Log.h>  // JLOG
 #include <beast/threads/Thread.h>
 #include <beast/cxx14/memory.h> // <memory>
 
 namespace ripple {
 namespace Resource {
 
-class ManagerImp
-    : public Manager
-    , public beast::Thread
+class ManagerImp : public Manager
 {
 private:
-    beast::Journal m_journal;
-    Logic m_logic;
+    beast::Journal journal_;
+    Logic logic_;
+    std::thread thread_;
+    std::atomic<bool> run_;
+
 public:
     ManagerImp (beast::insight::Collector::ptr const& collector,
         beast::Journal journal)
-        : Thread ("Resource::Manager")
-        , m_journal (journal)
-        , m_logic (collector, stopwatch(), journal)
+        : journal_ (journal)
+        , logic_ (collector, stopwatch(), journal)
+        , thread_ ()
+        , run_ (true)
     {
-        startThread ();
+        thread_ = std::thread {&ManagerImp::run, this};
     }
 
-    ~ManagerImp ()
+    ManagerImp () = delete;
+    ManagerImp (ManagerImp const&) = delete;
+    ManagerImp& operator= (ManagerImp const&) = delete;
+
+    ~ManagerImp () override
     {
-        stopThread ();
+        run_.store (false);
+        try
+        {
+            thread_.join();
```

Comment:
Yup.  That's my hope/intent as well.  However, I don't think the try/catch is a complete waste since our code does evolve over time.  But I don't think including the try/catch is very important, since any crash that would occur would likely be during shutdown anyway.

I'm guessing that your preference is that I remove the try/catch.  I'm happy to do so.  Am I guessing your preference correctly?  Thanks.